AK/LexicalPath.h

```diff
@@ -44,6 +44,30 @@ class LexicalPath {
         return LexicalPath { builder.to_string() };
     }
 
+    static String dirname(String path)
+    {
+        auto lexical_path = LexicalPath(move(path));
+        return lexical_path.dirname();
+    }
```

Comment:
Can we make the `LexicalPath` ctor take a `StringView`, and pass in a `StringView` here?
we're basically always copying the string anyway, so the `StringView` should avoid one copy

Reply:
Hold up, `StringView` incurs a guaranteed copy whereas a `String` copy may be a simple refcount bump.

Reply:
it'll always be a copy though, it goes through `canonicalized_path()`, which rebuilds the string...
oh it returns the same string if it matches some conditions, since that's "likely", let's assume that it's actually gonna happen a lot (I have my doubts, but whatever)